packages/client/src/components/Cart/CartTable/CartTable.component.js

```diff
@@ -1,9 +1,27 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './CartTable.styles.css';
 import CartSelectedProduct from '../CartSelectedProduct/CartSelectedProduct.component';
 import PropTypes from 'prop-types';
+import { useUserContext } from '../../../userContext';
 
-export default function CartTable({ products }) {
+export default function CartTable({ products, setProducts, isLoading }) {
+  const { user } = useUserContext();
+  const [selected, setSelected] = useState(false);
+
+  const handleChange = (product) => {
+    const productCart = [...products];
+
+    productCart.forEach((row, index) => {
+      if (productCart[index].id === product.id)
+        productCart[index].selected = !productCart[index].selected;
```

Comment:
This is a little out of scope, but in general, I would try to avoid mixing application state with UI state. 

In most component libraries you'll have a row expressed as this:
```
{
  selected: true,
  data: { /* Properties for a product */ }
}
```

So that you can make a clear separation between what is state for the table render, and what is actually the data to send to the API for example. This row state can safely be owned by the table component and be easily recreated if the `data` part changes.